lib/saml/responses/login.rb

```diff
@@ -7,6 +7,18 @@ module Responses
     class Login < OneLogin::RubySaml::Response
       include SAML::Responses::Base
 
+      def initialize(saml_response, options = {})
+        super(saml_response, options)
+      rescue OpenSSL::PKey::RSAError => e # "padding check failed." when decrypt fails
+        # :nocov: Temporary code only required during key rollovers using a new private key
+        raise e unless Settings.saml.key_new
+
+        duped_options = options.dup
+        duped_options[:settings].private_key = Settings.saml.key_new
+        super(saml_response, duped_options)
+        # :nocov:
```

Comment:
I took a crack at writing a spec to cover this code instead but decided against it because:
1. it requires keys, both invalid and valid one and we [consciously removed our localhost key](url)
2. it would also require an encrypted authresponse, which again, we cannot decrypt (see 1)
3. the code is meant to be temporary - it should only exist for the duration of keyrollver... probably 1-2 days
4. I tried using rspec mocks, `allow(OneLogin::RubySaml::Response).to receive(:new).and_raise(OpenSSL::PKey::RSAError)` - but this is effectively mocking `super` which doesn't really work.